scaffold/github/workflows/ComposerLockDiff.yml

```diff
@@ -17,46 +17,33 @@ jobs:
     steps:
       - uses: actions/checkout@v4
         with:
-          fetch-depth: 2
+          fetch-depth: 0
 
-      - uses: actions/cache@v4
+      - name: Check if composer.lock was changed
+        id: composer-lock-changed
+        uses: tj-actions/changed-files@v44
         with:
-          path: ${{ github.workspace }}/.ddev/.drainpipe-composer-cache
-          key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.lock') }}
-          restore-keys: |
-            ${{ runner.os }}-composer-
+          files: 'composer.lock'
 
-      - uses: ./.github/actions/drainpipe/set-env
-
-      - name: Install and Start DDEV
-        uses: ./.github/actions/drainpipe/ddev
+      - name: Delete sticky pull request comment
+        uses: marocchino/sticky-pull-request-comment@v2
         with:
-          git-name: Drainpipe Bot
-          git-email: no-reply@example.com
-          ssh-private-key: ${{ secrets.SSH_PRIVATE_KEY }}
-          ssh-known-hosts: ${{ secrets.SSH_KNOWN_HOSTS }}
-
-      - name: Build Project
-        run: ddev composer install
-
-      - name: Install composer lock diff
-        run: ddev composer global require davidrjonas/composer-lock-diff:^1.0
+          header: composer-lock-diff
+          delete: true
 
-      - name: Prepare repository
-        run: |
-          git reset --soft HEAD^1
-          git reset .
+      - name: Generate composer diff
+        if: ${{ steps.composer-lock-changed.outputs.any_changed == 'true' }}
+        id: composer-diff
+        uses: IonBazan/composer-diff-action@v1
+        with:
+          with-platform: true
+          with-links: true
```

Comment:
Does it make sense to also include the underlying CLI command in ddev via a Dockerfile? That way, developers are one step closer to debugging things when they go wrong.

Reply:
I don't really think so, and I would actually advocate for removing composer-lock-diff as a DDEV command. I never found a need for it to run this locally in the same way that it does with the action. I could just always debug the action itself or report an issue upstream with the community action.

It is worth noting this is a change from using https://github.com/davidrjonas/composer-lock-diff to https://github.com/IonBazan/composer-diff. There are some pros and cons here with that change:

| composer-lock-diff | composer-diff |
|-|-|
| Run `composer-lock-diff` | Run `composer diff` (composer plugin) |
| Last release Mar 2022 | Last release Dec 2023 |
| No GitHub action available | Maintained GitHub action available |
| Supports Drupal.org diffs | Does not yet support Drupal.org diffs (see https://github.com/IonBazan/composer-diff/issues/24) |

That said, having just the `composer diff` command available from the Dockerfile *was* useful, but I don't think it needs to be a task at all since it would be available as a native composer command.